src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,27 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  setEventListeners() {
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit();
+      this.close();
+    });
+    super.setEventListeners();
+  }
+
+  close() {
+    super.close();
+  }
```

Comment:
Please, delete `close`

You shouldn’t create the **same methods** of the parent class if you don’t change it inside and only call it with `super`. The parent methods are available in any inheritor of the class.